mintapi/cli.py

```diff
@@ -416,63 +413,83 @@ def main():
-        except Exception:
-            accounts = None
+    attention_msg = None
+    if options.attention:
+        attention_msg = mint.get_attention()
 
+    if options.accounts:
         try:
-            budgets = mint.get_budgets(limit=options.limit)
+            data = mint.get_account_data(limit=options.limit)
+            output_data(options, data, constants.ACCOUNT_KEY, attention_msg)
         except Exception:
-            budgets = None
+            data = None
 
-        data = {"accounts": accounts, "budgets": budgets}
-    elif options.budgets:
+    if options.budgets:
         try:
             data = mint.get_budgets(limit=options.limit)
+            output_data(options, data, constants.BUDGET_KEY, attention_msg)
         except Exception:
             data = None
     elif options.budget_hist:
         try:
             data = mint.get_budgets(limit=options.limit, hist=12)
+            output_data(options, data, constants.BUDGET_KEY, attention_msg)
         except Exception:
             data = None
-    elif options.accounts:
+
+    if options.transactions:
         try:
-            data = mint.get_account_data(limit=options.limit)
+            data = mint.get_transaction_data(
+                limit=options.limit,
+                start_date=options.start_date,
+                end_date=options.end_date,
+                include_investment=options.include_investment,
+                remove_pending=options.show_pending,
+            )
+            output_data(options, data, constants.TRANSACTION_KEY, attention_msg)
         except Exception:
             data = None
-    elif options.transactions:
-        data = mint.get_transaction_data(
-            limit=options.limit,
-            start_date=options.start_date,
-            end_date=options.end_date,
-            include_investment=options.include_investment,
-            remove_pending=options.show_pending,
-        )
-    elif options.categories:
-        data = mint.get_categories(
-            limit=options.limit,
-        )
-    elif options.investments:
-        data = mint.get_investment_data(
-            limit=options.limit,
-        )
-    elif options.net_worth:
-        data = mint.get_net_worth()
-    elif options.credit_score:
-        data = mint.get_credit_score()
-    elif options.credit_report:
-        data = mint.get_credit_report(
-            details=True,
-            exclude_inquiries=options.exclude_inquiries,
-            exclude_accounts=options.exclude_accounts,
-            exclude_utilization=options.exclude_utilization,
-        )
 
-    attention_msg = None
-    if options.attention:
-        attention_msg = mint.get_attention()
-    output_data(options, data, attention_msg)
+    if options.categories:
+        try:
+            data = mint.get_categories(
+                limit=options.limit,
+            )
+            output_data(options, data, constants.CATEGORY_KEY, attention_msg)
+        except Exception:
```

Comment:
Does this make sense to replicate any more? Is setting `data = None` providing any utility, or blindly catching all exceptions? I wonder if we could simply remove these calls from the try/except and tidy it up?

Reply:
Very valid point. I have a desire to refactor it a little more once it's out in the wild. 

But yeah it's really just blindly catching exceptions and initializing the variable, though probably unnecessary.